src/aiokem/main.py

```diff
@@ -1,3 +1,92 @@
-def add(n1: int, n2: int) -> int:
-    """Add the arguments."""
-    return n1 + n2
+"""AioKem class for interacting with Kohler Energy Management System (KEM) API."""
+
+from typing import Any
+
+from aiohttp import ClientSession, ClientTimeout
+
+AUTHENTICATION_URL = "https://kohler-homeenergy.okta.com/oauth2/default/v1/token"
+CLIENT_KEY = (
+    "MG9hMXFpY3BkYWdLaXdFekYxZDg6d3Raa1FwNlY1T09vMW9"
+    "PcjhlSFJHTnFBWEY3azZJaXhtWGhINHZjcnU2TWwxSnRLUE5obXdsMEN1MGlnQkVIRg=="
+)
+API_KEY = "pgH7QzFHJx4w46fI~5Uzi4RvtTwlEXp"
+API_BASE = (
+    "https://az-amer-prod-hems-capp.jollyglacier-72224ec0.eastus.azurecontainerapps.io"
+)
+
+
+class AioKem:
+    """AioKem class for interacting with Kohler Energy Management System (KEM) API."""
+
+    def __init__(self, session: ClientSession = None) -> None:
+        """Initialize the AioKem class."""
+        self._token: str | None = None
+        self._refresh_token: str | None = None
+        self._session = (
+            session if session else ClientSession(timeout=ClientTimeout(total=10))
+        )
+
+    async def login(self, username: str, password: str) -> None:
+        """Login to the server."""
+        headers = {
+            "accept": "application/json",
+            "authorization": f"Basic {CLIENT_KEY}",
+            "content-type": "application/x-www-form-urlencoded",
+        }
+        data = {
+            "grant_type": "password",
+            "username": username,
+            "password": password,  # Replace 'pass' with the actual password
+            "scope": "openid profile offline_access email",
+        }
+
+        response = await self._session.post(
+            AUTHENTICATION_URL, headers=headers, data=data
```

Comment:
note that passing a `str` url to `aiohttp` has to convert it to `yarl.URL` every time so its better to pass it a `URL` if we already have it to avoid the conversion each request